### PR TITLE
Alex

### PR DESCRIPTION
**Pull request by cag472**

### CODE REVIEW

File: FakeRate/app_region/SS.cc [hunk 1]
```diff
@@ -0,0 +1,138 @@
+#include "SS.h"
```
Comment (cerati) [inline FakeRate/app_region/SS.cc:1]: I don't understand why adding this file is needed. It causes massive changes from ss. to ss:: that are very confusing for reviewing diffs and make merges of other commits difficult. Also if we really need this, we should remove the old lepfilter.h/cc

File: FakeRate/app_region/ScanChain.C [hunk 2]
```diff
@@ -512,147 +523,170 @@ int ScanChain( TChain* chain, TString fakeratefile, TString option = "", TString
...
+		  pTrelvsMiniIso_histo_mu->Fill( std::min(ss::lep2_miniIso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
 		  hists[getHist("pTrel_histo_mu")]->Fill(std::min(lep2_ptrel_v1,float(29.9)) );
 		}
 	      }
 	    }
 	  }
 
-  if (ss.hyp_class() == 3){
+
+  if (ss::hyp_class() == 3 && lep1_pT >= 20 && lep2_pT >= 20){
```
Comment (cerati) [inline FakeRate/app_region/ScanChain.C]: why >20 for both legs? it won't work for LL and HL

File: FakeRate/app_region/ScanChain.C [hunk 3]
```diff
@@ -512,147 +523,170 @@ int ScanChain( TChain* chain, TString fakeratefile, TString option = "", TString
...
 
     counter++;
 
-    if (ss.lep1_passes_id()==0) continue;
-    if (ss.lep2_passes_id()==0) continue;
-
```
Comment (cerati) [inline FakeRate/app_region/ScanChain.C:553]: why are these removed? we want the observed, so both leptons must pass the selection (ok, maybe it is redundant)

File: FakeRate/app_region/ScanChain.C (hunk 2 above)
Comment (cag472) [inline FakeRate/app_region/ScanChain.C]: Hi Giuseppe,

About the 20/20 - I agree, this whole change is unnecessary, will revert
it..

About the class files -- this is just allowing this directory to use the
same class files as everywhere else.

Maybe we should move this discussion to private e-mail, I think these
comments are going to the whole SNT list !!

-Alex
